File: week 2/pset/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(){

    string text = get_string("Text: ");

    int total_letters = count_letters(text);
    int total_words = count_words(text);
    int total_sentences = count_sentences(text);

    printf("%i, %i, %i\n", total_letters, total_words, total_sentences);

    float L = (total_letters / (float) total_words) * 100;

    float S = (total_sentences / (float) total_words) * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if(index < 1){
        printf("Before Grade 1\n");
    }
    else if(index >= 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", index);
    }

}

int count_letters(string text){

    int letters = 0;

    for(int i = 0; i <= strlen(text); i++){

        if(isalpha(text[i]) && !isspace(text[i])){

            letters++;

        }

    }

    return letters;

}

int count_words(string text){

    int words = 0;

    for(int i = 0; i <= strlen(text); i++){

        if(isspace(text[i])){

            words++;

        }

    }

    return words + 1;

}

int count_sentences(string text){

    int sentences = 0;

    for(int i = 0; i <= strlen(text); i++){

        if(text[i] == '.' || text[i] == '!' || text[i] == '?'){

            sentences++;

        }

    }

    return sentences;

}

File: week 6/pset/readability/readability.py
from cs50 import get_string

def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha() and not(text[i].isspace()):
            letters += 1
    return letters

def count_words(text):
    words = 0
    for i in range(len(text)):
        if text[i].isspace():
            words += 1
    return words + 1

def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?') :
            sentences += 1
    return sentences




text = get_string("Enter your Text: ")

total_letters = count_letters(text)
total_words = count_words(text)
total_sentences = count_sentences(text)

L = (total_letters / total_words * 100)
S = (total_sentences / total_words * 100)
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(index))


